### ISSUE TITLE
Apply OverloadMethodsDeclarationOrder check for the same access modificators only

### ISSUE THREAD

**Issue by ychulovskyy** [checkstyle/checkstyle#3235]
Assume we have the following code:

```
@Override
public void foo(String s) {}
public void notFoo() {} // Have to be after foo(int i, String s)
private void foo(int i, String s) {}
```

OverloadMethodsDeclarationOrder check will find violation here since notFoo() is between 2 methods with the same name  "foo".

We would like to group methods by access modifiers.
That is why it's OK for us that there is notFoo() between 2 foo() methods, since first foo() is public but second foo() is private.

It would be great to add "applyForTheSameAccessLevelOnly" property to OverloadMethodsDeclarationOrder to apply the check to methods with the same access modifiers only.
E.g. If both foo() methods are public - use the check.
If one of foo() methods is private but another one is private - don't apply the OverloadMethodsDeclarationOrder check.
##

**Comment by romani** [comment]
please update description to follow standard issue report - http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by romani** [comment]
closed in favor of #3770
